buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/LicenseHeadersTask.groovy

```diff
@@ -33,8 +36,15 @@ import java.nio.file.Files
  */
 public class LicenseHeadersTask extends AntTask {
 
+    // checks all the java sources (allJava)
+    private Iterable<File> input = project.sourceSets.collect({it.allJava.srcDirs}).flatten()
```

Comment:
I'm still apprehensive about this, since it means the srcDirs will be read at task construction time, and also at ctor time used to find all source files. It will work now, since we check in all generated files. I would at least move this into the ctor in an afterEvaluate, so that if srcDirs are modified, it will pick up the right ones. And I would also add a comment explaining the limitation I mentioned here.